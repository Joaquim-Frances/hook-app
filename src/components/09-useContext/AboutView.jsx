import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const AboutView = () => {
    const {user, setUser} = useContext(UserContext);

    return (
        <div>
            <h1>About.</h1>
            <hr/>

            <pre>
                {JSON.stringify(user)}
            </pre>

            <button className='btn btn-primary' onClick={()=>setUser(null)}>
                Logout
            </button>
        </div>
    )
}
