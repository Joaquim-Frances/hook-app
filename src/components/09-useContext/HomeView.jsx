import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const HomeView = () => {

    const {user} = useContext(UserContext);

    console.log(user);

    return (
        <div>
            <h1>Home.</h1>
            <hr/>

            <pre className='container'>
                {JSON.stringify(user, null, 3)}
            </pre>
        </div>
    )
}
