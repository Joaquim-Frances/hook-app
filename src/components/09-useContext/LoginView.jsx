import React, { useContext } from 'react';
import { UserContext } from './UserContext';


export const LoginView = () => {
    const { user, setUser } = useContext(UserContext)


    const login ={
        id: 123,
        userName: 'Quim',
    }

    return (
        <div>
            <h1>Login.</h1>
            <hr/>
        
            <button className='btn btn-primary' onClick={()=>setUser(login)}> Login </button>
            <pre>{JSON.stringify(user)}</pre>
        </div>
    )
}
