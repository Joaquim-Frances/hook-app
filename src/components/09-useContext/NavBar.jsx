import React from 'react'
import { Link } from 'react-router-dom'

export const NavBar = () => {
    return (
        <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
            <div className="container-fluid">
                <a className="navbar-brand">useContext</a>
                
                <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                <div className="navbar-nav">
                    <Link to='/'><a className="nav-link active">Home</a></Link>
                    <Link to='/login'><a className="nav-link active">Login</a></Link>
                    <Link to='/about'><a className="nav-link active">About</a></Link>
                </div>
                </div>
            </div>
        </nav>

    )
}
