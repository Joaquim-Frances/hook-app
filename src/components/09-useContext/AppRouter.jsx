import React from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect,
  } from "react-router-dom";
import { AboutView } from './AboutView';
import { HomeView } from './HomeView';
import { LoginView } from './LoginView';
import { NavBar } from './NavBar';

export const AppRouter = () => {
    return (
        <Router>
            <div>
                <NavBar />
                <div className="container">
                    <Switch>
                        <Route exact path="/" component={ HomeView } />
                        <Route exact path="/about" component={ AboutView } />
                        <Route exact path="/login" component={ LoginView } />
                        <Redirect to="/" />
                    </Switch> 
                </div>
            </div>
        </Router>


        
    )
}
