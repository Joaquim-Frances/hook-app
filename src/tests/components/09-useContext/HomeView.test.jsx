import { mount, shallow } from 'enzyme';
import React from 'react';
import { HomeView } from '../../../components/09-useContext/HomeView';
import { UserContext } from '../../../components/09-useContext/UserContext';


describe('Pruebas en HomeView', () => {

    const user = {
        name: 'Quim',
        email: 'quimifrances@gmaill.com'
    }

    const wrapper = shallow(  //mount no funciona
    <UserContext.Provider value={{user}}>
    <HomeView />
    </UserContext.Provider>
    );

    test('should display correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
    
})
