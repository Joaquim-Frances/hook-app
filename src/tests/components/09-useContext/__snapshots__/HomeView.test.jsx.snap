// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en HomeView should display correctly 1`] = `<HomeView />`;
